internal/blockprocessor/committer.go

```diff
@@ -450,7 +449,18 @@ func constructDBEntriesForConfigTx(tx *types.ConfigTx, oldConfig *types.ClusterC
 		return nil, err
 	}
 
-	newConfigSerialized, err := proto.Marshal(tx.NewConfig)
+	// calculate the new max RaftID from the peers and old config
+	maxID := oldConfig.GetConsensusConfig().GetRaftConfig().GetMaxRaftId()
+	// avoid mutating the tx, as it continues to be processed later
+	newConfigClone := proto.Clone(tx.NewConfig).(*types.ClusterConfig)
+	for _, m := range newConfigClone.GetConsensusConfig().GetMembers() {
+		if id := m.GetRaftId(); id > maxID {
+			maxID = id
+		}
+	}
+	newConfigClone.ConsensusConfig.RaftConfig.MaxRaftId = maxID
```

Comment:
The `ConsensusConfig.RaftConfig.MaxRaftID` value is used my the admin to set a unique number, say `ConsensusConfig.RaftConfig.MaxRaftID+1`. It will be updated automatically if the config-tx commits as valid.